Adds allow_look option to EvMenu.

shollen opened the pull request:
Solves the issue of the default EvMenu 'look' command interfering with custom code for permutations of ‘look’ in a default node.

I added an allow_look option similar to the allow_quit option, but without the option of a replacement function. allow_look defaults to True.

I implemented this change when I found that I couldn't use the single letter 'l' as input  in a _default node asking for a character's dexterity orientation, as the code uses 'left'.startswith and 'l' is an alias for the default 'look' command.

Griatch commented:
@shollen 

Thanks for the PR! Terribly sorry but I interpreted your latest mailing list post as if you weren't  going to move forward with this - so I added the functionality in the wclient branch a few hours ago (you can see the update in issue #949). I should have commented in the issue as well though, to make sure you saw it. Your PR looks fine and uses correct branching and everything so now I feel bad. :-(

I did add few more changes though (which will not show until wclient branch merges). Notably I changed the name of `allow_quit` to `auto_quit` (since the effect of it is to automatically add a quit command). That also made more sense when adding additional `auto_look` and `auto_help` keywords (`auto_help` will turn off the automatic use of "h" and "help" which I suppose could have been a similar problem as for "l" and "look" ). 

...So I'm closing this for now (you can still comment if you want), hope it doesn't discourage you from future PRs! 
.
Griatch

shollen commented:
I’m not just new to Python and Evennia. I’ve never used Git before either, so it took me a bit to get my local stuff configured correctly and set up to do things the right way. I love Python. I love Evennia. And I want to help the codebase grow however I can. But game design and development is in my blood, so my focus is finding/building the best tools I can to get the job done. I learned a ton about Git by getting the EvMenu stuff pushed. The issue has been fixed. Doesn’t matter to me whose code it is.  :-)  So don’t feel bad! I learned!

> On Apr 17, 2016, at 1:32 AM, Griatch <notifications@github.com> wrote:
> 
> @shollen <https://github.com/shollen>
> Thanks for the PR! Terribly sorry but I interpreted your latest mailing list post as if you weren't going to move forward with this - so I added the functionality in the wclient branch a few hours ago (you can see the update in issue #949 <https://github.com/evennia/evennia/issues/949>). I should have commented in the issue as well though, to make sure you saw it. Your PR looks fine and uses correct branching and everything so now I feel bad. :-(
> 
> I did add few more changes though (which will not show until wclient branch merges). Notably I changed the name of allow_quit to auto_quit (since the effect of it is to automatically add a quit command). That also made more sense when adding additional auto_look and auto_help keywords (auto_help will turn off the automatic use of "h" and "help" which I suppose could have been a similar problem as for "l" and "look" ).
> 
> ...So I'm closing this for now (you can still comment if you want), hope it doesn't discourage you from future PRs! 
> .
> Griatch
> 
> —
> You are receiving this because you were mentioned.
> Reply to this email directly or view it on GitHub <https://github.com/evennia/evennia/pull/959#issuecomment-210959128>
